python/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+use pyo3::{
+    exceptions::{PyException, PyIOError, PyValueError},
+    prelude::*,
+    types::PyDict,
+};
+use std::path::PathBuf;
+
+/// A CQL2 expression.
+#[pyclass]
+struct Expr(::cql2::Expr);
+
+/// A SQL query
+#[pyclass]
+struct SqlQuery {
+    #[pyo3(get)]
+    query: String,
+
+    #[pyo3(get)]
+    params: Vec<String>,
+}
+
+#[pymethods]
+impl Expr {
+    /// Parses a CQL2 expression from a file path.
+    #[staticmethod]
+    fn from_path(path: PathBuf) -> PyResult<Expr> {
+        ::cql2::parse_file(path).map(Expr).map_err(to_py_error)
```

Comment:
Especially if this library grows larger, you can create a custom error type so that you don't have to use `.map_err` on every method. https://github.com/kylebarron/arro3/blob/main/pyo3-arrow/src/error.rs

Reply:
Yeah good point — I hadn't seen the "wrap a `PyErr`" pattern before, makes sense. I _think_ for this one it's not worth it yet, because so many of the methods are just direct calls down to the `cql2` crate, meaning we have to map the error anyways. But agreed that if we do more we should simplify the error handling.